compiler/src/dotty/tools/dotc/ast/NavigateAST.scala

```diff
@@ -141,7 +141,9 @@ object NavigateAST {
           case _ =>
         val iterator = p match
           case defdef: DefTree[?] =>
-            p.productIterator ++ defdef.mods.productIterator
+            val mods = defdef.mods.productIterator
+            val annotations = defdef.symbol.annotations.filter(_.tree.span.contains(span)).map(_.tree).iterator
```

Comment:
```suggestion
            val annotations = defdef.symbol.annotations.iterator.filter(_.tree.span.contains(span)).map(_.tree)
```
otherwise I think the iterator isn't that useful.

Reply:
That's right, the iterator isn't useful there.